trestlebot/entrypoints/sync_upstreams.py

```diff
@@ -0,0 +1,131 @@
+        """Initialize."""
+        # Setup base arguments
+        super().__init__(parser)
+        self.setup_sync_upstreams_arguments()
+
+    def setup_sync_upstreams_arguments(self) -> None:
+        """Setup arguments for the sync upstreams entrypoint."""
+        self.parser.add_argument(
+            "--sources",
+            type=str,
+            required=True,
+            help="Comma-separated list of upstream git sources to fetch from. Each source is a string \
+                of the form <repo_url>@<ref> where ref is a git ref such as a tag or branch.",
+        )
+        self.parser.add_argument(
+            "--include-model-names",
+            type=str,
+            required=False,
+            help="Comma-separated list of glob patterns for model names to include when running \
+                tasks (e.g. --include-models=component_x,profile_y*)",
+        )
+        self.parser.add_argument(
+            "--exclude-model-names",
+            type=str,
+            required=False,
+            help="Comma-separated list of glob patterns for model names to exclude when running \
+                tasks (e.g. --exclude-models=component_x,profile_y*)",
+        )
+        self.parser.add_argument(
+            "--skip-validation",
+            action="store_true",
+            help="Skip validation of the models when they are copied",
+        )
+
+    def run(self, args: argparse.Namespace) -> None:
+        """Run the sync upstreams entrypoint."""
+        exit_code: int = SUCCESS_EXIT_CODE
+        try:
+            set_log_level_from_args(args)
+            if not args.sources:
+                raise EntrypointInvalidArgException(
+                    "--sources", "Must set at least one source to sync from."
+                )
+
+            # Assume that if exclude_model_names is not set, then
+            # skip nothing and if include_model_names is not set, then include all.
+            include_model_list: List[str] = ["*"]
+            if args.include_model_names:
+                include_model_list = comma_sep_to_list(args.include_model_names)
+            model_filter: ModelFilter = ModelFilter(
+                skip_patterns=comma_sep_to_list(args.exclude_model_names),
+                include_patterns=include_model_list,
+            )
+
+            validate: bool = not args.skip_validation
+
+            sync_upstreams_task: TaskBase = SyncUpstreamsTask(
+                working_dir=args.working_dir,
+                git_sources=comma_sep_to_list(args.sources),
+                model_filter=model_filter,
+                validate=validate,
+            )
+            pre_tasks: List[TaskBase] = [sync_upstreams_task]
+
+            super().run_base(args, pre_tasks)
+        except Exception as e:
+            exit_code = handle_exception(e)
+
+        sys.exit(exit_code)
+
+
+def main() -> None:
+    """Run the CLI."""
+    parser = argparse.ArgumentParser(
```

Comment:
just an observation about developing CLI - we have found https://palletsprojects.com/p/click/ to be a higher starting point when developing CLIs... not suggesting to refactor but thought you might be interested

Reply:
Thanks for the suggestion! I think `click` could be really useful if/when we decide that we need to start grouping commands for ease and discovererabilty. So far they have been standalone/dissimilar so we have keep it is as a flat structure.